Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
+	{
+		INI_Reader ini;
+
+		string factionpropfile = "..\\data\\initialworld.ini";
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Group"))
+				{
+					uint ids_name;
+					string nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("nickname"))
+						{
+							nickname = ini.get_value_string();
+						}
+						else if (ini.is_value("ids_name"))
+						{
+							ids_name = ini.get_value_int(0);
+						}
+
+					}
+					factions[nickname] = ids_name;
+				}
+			}
+			ini.close();
+			ConPrint(L"Resetrep: Loaded %u factions\n", factions.size());
+		}
+	}
+	
+	map<wstring, uint> MiscCmds::Resetrep_load_Time_limits_for_player_account(string filename)
+	{
+		INI_Reader ini;
+		map<wstring, uint> tempmap;
+		if (ini.open(filename.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Playership"))
+				{
+					uint lastresettime = 0;
+					wstring nickname = L"0";
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("lastresettime"))
+						{
+							lastresettime = ini.get_value_int(0);
+						}
+					}
+
+					tempmap[nickname] = lastresettime;
+				}
+			}
+			ini.close();
+		}
+		return tempmap;
+	}
+
+	void MiscCmds::Resetrep_save_Time_limits_to_player_account(string filename, map<wstring, uint> tempmap)
+	{
+
+		FILE *file = fopen(filename.c_str(), "w");
+		if (file)
+		{
+			for (map<wstring, uint>::iterator i = tempmap.begin();
+				i != tempmap.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
+				{
+					wstring temp = i->first;
+					fprintf_s(file, "[Playership]\n");
+					ini_write_wstring(file, "shipname", temp);
+					fprintf_s(file, "lastresettime = %d\n", i->second);
+				}
+			}
+			fclose(file);
+		}
+	}
+
+	/** A command to reset all reputations to zero that you might have */
+	bool MiscCmds::UserCmd_ResetRep(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+	{
+		HK_ERROR err;
+
+		wstring wscCharname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+
+		uint iShip;
+		pub::Player::GetShip(iClientID, iShip);
+		if (iShip)
+		{
+			PrintUserCmdText(iClientID, L"ERR Not in base");
+			return true;
+		}
+
+		// Read the current number of credits for the player
+		// and check that the character has enough cash.
+		int iCash = 0;
+		if ((err = HkGetCash(wscCharname, iCash)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR (1) %s", HkErrGetText(err).c_str());
+			return true;
+		}
+
+		if (set_iResetrepCost > 0 && iCash < set_iResetrepCost)
+		{
+			PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+			return true;
+		}
+
+
+		wstring wscCharFileName;
+		if ((err = HkGetCharFileName(wscCharname, wscCharFileName)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR (2)" + HkErrGetText(err));
```

Comment:
why do we concatenate here but do string formatting and c_str above? also missing space after `(2)`?

Reply:
>I still don't understand why we're checking the current time inside a function intended to write a map to disk.
This was intended... to write to disk only time limits which have not expired yet

Reply:
renamed error